src/cmd/ksh93/tests/functions.sh

```diff
@@ -268,9 +268,11 @@ chmod +x $tmp/foobar
 FPATH=$tmp
 cd $tmp
 print 'function zzz { return 0;}' > zzz
-PATH=$PATH:
+# Remove other commands named zzz from PATH.
+old_path=$PATH
```

Comment:
I think it will be better to just change name of the function then saving and restoring old path.

Reply:
I had posted in #429 a diff that renames `zzz` to `zzf`. That doesn't fix the problem, because there might be other commands named `zzf` in PATH.

Reply:
I agree with @kernigh. The unit tests have to assume there could be arbitrary external commands with names that conflict with ksh functions created by a test. However, I just checked and neither ksh93u nor this project's version runs the external command if I create a function with the same name in an interactive shell. Which is the expected, correct, behavior. It shouldn't be necessary to manipulate `PATH` in this manner. So what is different about the unit test environment that requires this hack?

Reply:
I looked at *functions.sh* and see that the problem isn't due to the function being named `zzz`. It's because the function is put in a file named `zzz` which is an external command found later in `PATH` than the system provided command of the same name. So there isn't any need to save/restore `PATH`. We simply need to ensure the temp dir that is created, and which is the CWD when the test runs, is at the front of `PATH`. And that should be done for every unit test. Just as every unit test creates a temp dir and cd's to it. So rather than merge this change, and more generally do this sort of thing in each individual unit test, we should create a standard way to have a preamble and postscript for each unit test. I'll take care of that as the current situation bugs me.